File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.java
class Solution {
    public int[] sortedSquares(int[] nums) {
        
        int front = 0;
        int rear = nums.length - 1;
        int[] ans = new int[nums.length];
        int idx = nums.length - 1;
        while(front <= rear){
            if(Math.abs(nums[front]) < Math.abs(nums[rear])){
                ans[idx--] = nums[rear] * nums[rear];
                rear--;
            }else{
                ans[idx--] = nums[front] * nums[front];
                front++;
            }
        }
        
        
        return ans;
        
    }
}

File: 977-squares-of-a-sorted-array/977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        ans = [0 for _ in range(len(nums))]
        left = 0
        right = len(nums) - 1
        idx = len(nums)
        while left <= right:
            idx-=1
            if abs(nums[left]) > abs(nums[right]):
                ans[idx] = nums[left]**2
                left+=1
            else:
                ans[idx] = nums[right]**2
                right-=1
        return ans
        
        
        
        